File: typescript-client-sdk/src/sdk/models/operations/getonecrmnote.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetOneCrmNoteQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=id" })
  id: string;
}


export class GetOneCrmNoteResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=note" })
  note?: shared.Note;
}


export class GetOneCrmNoteRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetOneCrmNoteQueryParams;
}


export class GetOneCrmNoteResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetOneCrmNoteResponseBody;
}

File: go-client-sdk/pkg/models/operations/getonecrmnote.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetOneCrmNoteQueryParams struct {
	AccessToken string `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool  `queryParam:"style=form,explode=true,name=allFields"`
	ID          string `queryParam:"style=form,explode=true,name=id"`
}

type GetOneCrmNoteResponseBody struct {
	Note *shared.Note `json:"note,omitempty"`
}

type GetOneCrmNoteRequest struct {
	QueryParams GetOneCrmNoteQueryParams
}

type GetOneCrmNoteResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetOneCrmNoteResponseBody
}
